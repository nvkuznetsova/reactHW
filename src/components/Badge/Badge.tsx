import React from 'react';

export type BadgeProps = {
  badgeText: string,
  badgeClassName: string,
};

export const Badge: React.FC<BadgeProps> = ({ badgeText, badgeClassName }) => (
  <span className={`badge badge-pill ml-1 ${badgeClassName}`}>{badgeText}</span>
);