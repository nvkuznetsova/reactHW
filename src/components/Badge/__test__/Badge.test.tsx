import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Badge, BadgeProps } from '../Badge';

describe('Badge', () => {
  const getProps = (): BadgeProps => ({
    badgeText: 'badge',
    badgeClassName: 'class',
  });
  const setup = (): ShallowWrapper => shallow(<Badge {...getProps()} />);

  it('should match snapshot', () => {
    const component = setup();
    expect(component).toMatchSnapshot()
  });
});