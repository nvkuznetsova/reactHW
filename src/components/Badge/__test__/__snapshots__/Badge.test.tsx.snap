// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<span
  className="badge badge-pill ml-1 class"
>
  badge
</span>
`;
